cmds/exp/smbios_transfer/smbios_transfer.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// smbios_transfer sends SMBIOS tables to BMC through IPMI blob interfaces.
+//
+// Synopsis:
+//     smbios_tranfer [-num_retries]
+//
+// Options:
+//	-num_retries: number of times to retry transferring SMBIOS tables
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+
+	"github.com/u-root/u-root/pkg/ipmi"
+	"github.com/u-root/u-root/pkg/ipmi/blobs"
+)
+
+const (
+	maxWriteSize uint32 = 128
+
+	// IPMI blob ID on BMC
+	smbiosBlobID = "/smbios\x00"
+
+	sysfsPath = "/sys/firmware/dmi/tables"
+)
+
+var (
+	retries = flag.Int("num_retries", 2, "Number of times to retry transferring SMBIOS tables")
+)
+
+func writeCommitSmbiosBlob(id string, data []uint8, h *blobs.BlobHandler) (rerr error) {
+	sessionID, err := h.BlobOpen(id, blobs.BMC_BLOB_OPEN_FLAG_WRITE)
+	if err != nil {
+		return fmt.Errorf("IPMI BlobOpen for %s failed: %v", id, err)
+	}
+	defer func() {
+		// If the function returned successfully but failed to close the blob,
+		// return an error.
+		if err := h.BlobClose(sessionID); err != nil && rerr == nil {
+			rerr = fmt.Errorf("IPMI BlobClose %s failed: %v", id, err)
```

Comment:
What if there are two errors? If there is an error here will this return wipe out the other error value?

Reply:
I will push a new change that will concatenate them if two errors. 
`
	defer func() {
		// If the function returned successfully but failed to close the blob,
		// return an error.
		if err := h.BlobClose(sessionID); err != nil {
			err = fmt.Errorf("IPMI BlobClose %s failed: %v", id, err)
			if rerr != nil {
				rerr = fmt.Errorf("%v; %v", rerr, err)
				return
			}
			rerr = err
		}
	}()
`